### PR TITLE
(SERVER-2265) Refactor utils, configs, actions into separate dirs

### PR DESCRIPTION
**Pull request by lazartea**
This commit separates the utils, configs, and actions into their own directories.
It also renames files to be more consistant.

### CODE REVIEW

File: lib/puppetserver/ca/utils/parse.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
+module Puppetserver
+  module Ca
+    module Utils
+      module Parse
```
Comment (lazartea) [inline lib/puppetserver/ca/utils/parse.rb]: How do we feel about this name change? I figured ``Puppetserver::Ca::Utils::Utils`` didn't make much sense.

File: lib/puppetserver/ca/actions/clean.rb [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+require 'puppetserver/ca/utils/file_system'
+require 'puppetserver/ca/config/puppet'
+require 'puppetserver/ca/actions/revoke'
+
+require 'optparse'
+require 'json'
+
+module Puppetserver
+  module Ca
+    module Actions
```
Comment (Magisus) [inline lib/puppetserver/ca/actions/clean.rb]: For namespacing like this, should it be `Action` or `Actions`?

File: lib/puppetserver/ca/utils/config.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module Puppetserver
+  module Ca
+    module Utils
```
Comment (Magisus) [inline lib/puppetserver/ca/utils/config.rb]: How do people feel about putting all parent modules on one line to reduce indentation, e.g. 
```
module Puppetserver::Ca::Utils
  module Config
  end
end
```

File: lib/puppetserver/ca/utils/parse.rb (hunk 1 above)
Comment (Magisus) [reply lib/puppetserver/ca/utils/parse.rb]: I would prefer something like `Cli` maybe.

File: lib/puppetserver/ca/actions/clean.rb [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      class Clean
+
+        include Puppetserver::Ca::Utils::Parse
```
Comment (justinstoller) [inline lib/puppetserver/ca/actions/clean.rb]: by including the whole module you have access to the methods defined in it w/o having to namespace them.

I think we should just include the `Puppetserver::Ca::Utils` module, we can then directly refer to constants defined within it (eg we can just use `Parse.parse_with_errors` below)

File: lib/puppetserver/ca/actions/clean.rb (hunk 2 above)
Comment (justinstoller) [reply lib/puppetserver/ca/actions/clean.rb]: I think the coherence of the object within the namespace helps. Like I think of `Utils::FileSystem` as a collection of utility methods for interacting with the filesystem. That kinda breaks down with `HttpClient` as a singular entity vs a collection of methods. With `Actions::Clean` I would expect to be a collection of methods for cleaning vs `Action::Clean` as a singular clean action. I guess its really that I think of the parent module as providing some description of the child object vs describing itself...

tl;dr: I'd go with `Action`, and I could go either way on `Utils`.

File: lib/puppetserver/ca/utils/config.rb (hunk 3 above)
Comment (justinstoller) [reply lib/puppetserver/ca/utils/config.rb]: no feelings either way

File: lib/puppetserver/ca/utils/parse.rb (hunk 1 above)
Comment (justinstoller) [reply lib/puppetserver/ca/utils/parse.rb]: I wouldn't go with just `Cli` since there's a class named that in the higher level `Puppetserver::Ca` module. But maybe `CliParsing` or something?

**Comment by lazartea** [comment]
@justinstoller is this still broken? When you look at individual commits, you can see that each file was renamed, and then changed. But when we look at the total changes, it's showing deleted files again... Will we still retain the git history?

**Comment by justinstoller** [comment]
It looks great, though we should have `(SERVER-2265)` at the start of each commit

File: lib/puppetserver/ca/utils/config.rb (hunk 3 above)
Comment (justinstoller) [reply lib/puppetserver/ca/utils/config.rb]: We played with this and combining the names messed with Ruby's constant lookup so we went with the old way.

**Comment by lazartea** [comment]
@Magisus @justinstoller fixed rebase issue, and tested locally. Looks like it works!
